src/extension/node/commands/edit_connections.ts

```diff
@@ -181,30 +61,26 @@ async function handleConnectionsPreSave(
   for (let index = 0; index < connections.length; index++) {
     const connection = connections[index];
     connection.isDefault = index === defaultIndex;
-    if (connection.backend === ConnectionBackend.Postgres) {
+    if ('password' in connection) {
       if (connection.useKeychainPassword === false) {
         connection.useKeychainPassword = undefined;
         await deletePassword(
           'com.malloy-lang.vscode-extension',
           `connections.${connection.id}.password`
         );
       }
-    }
-    if (
-      connection.backend === ConnectionBackend.Postgres &&
-      connection.password !== undefined &&
-      connection.password !== ''
-    ) {
-      modifiedConnections.push({
-        ...connection,
-        password: undefined,
-        useKeychainPassword: true,
-      });
-      await setPassword(
-        'com.malloy-lang.vscode-extension',
-        `connections.${connection.id}.password`,
-        connection.password
-      );
+      if (connection.password) {
+        modifiedConnections.push({
+          ...connection,
+          password: undefined,
```

Comment:
I'll take care of merging my password logic with this, right now does not really use keychain.

Reply:
We should probably be using the VS Code secret API, since keytar is deprecated.